agent/run_job.go

```diff
@@ -166,6 +161,87 @@ func (r *JobRunner) Run(ctx context.Context) error {
 	return nil
 }
 
+func (r *JobRunner) validateConfigAllowlists(job *api.Job) error {
+	validations := map[string]func() error{
+		"repo": func() error {
+			return validateJobValue(r.conf.AgentConfiguration.AllowedRepositories, job.Env["BUILDKITE_REPO"])
+		},
+		"environment variables": func() error {
+			return validateEnv(job.Env, r.conf.AgentConfiguration.AllowedEnvironmentVariables)
+		},
+		"plugins": func() error {
+			return r.validatePlugins()
+		},
+	}
+
+	for name, validation := range validations {
+		if err := validation(); err != nil {
+			return fmt.Errorf("failed to validate %s: %w", name, err)
+		}
+	}
```

Comment:
A map of validations and a loop feels like an excessive amount of mechanism when there's only three of them...

```suggestion
	if err := validateJobValue(r.conf.AgentConfiguration.AllowedRepositories, job.Env["BUILDKITE_REPO"]); err != nil {
		return fmt.Errorf("failed to validate BUILDKITE_REPO: %w", err)
	}
	if err := validateEnv(job.Env, r.conf.AgentConfiguration.AllowedEnvironmentVariables); err != nil {
		return fmt.Errorf("failed to validate environment variables: %w", err)
	}
	if err := r.validatePlugins(); err != nil {
		return fmt.Errorf("failed to validate plugins: %w", err)
	}
```

Reply:
It's consistent with WET vs DRY. I'll allow it.

Reply:
Then at least:
```go
"plugins": r.validatePlugins,
```